syntax = "proto3";

import "library/cpp/getoptpb/proto/confoption.proto";

message TConfig {
    option (NGetoptPb.Title) = "BertTfApplier";
    string YtProxy = 1;
    string InputTable = 2;
    string OutputTable = 3;
    string InputTextColumn = 4;
    optional string OutputSentenceEmbeddingColumn = 5;
    optional string OutputWordEmbeddingsColumn = 6;
    optional string OutputWordsColumn = 7;
    optional int32 BatchSize = 8;
    optional int32 MemoryLimitMB = 9;
    optional int32 DataSizePerJobKB = 10;
    string StartTriePath = 11;
    string ContinuationTriePath = 12;
    string VocabFilePath = 13;
    string GraphDefPath = 14;
    int32 SequenceLength = 15;
    string TokenEmbeddingsOutput = 16;
    string SentenceEmbeddingOutput = 17;
    int32 NumInterOpThreads = 18;
    int32 NumIntraOpThreads = 19;
    int32 CudaDeviceIdx = 20;
}
